windows/src/engine/keyman32/keyboardoptions.cpp

```diff
@@ -182,3 +206,68 @@ void IntSaveKeyboardOption(LPCSTR key, LPINTKEYBOARDINFO kp, int nStoreToSave)
     r.WriteString(kp->Keyboard->dpStoreArray[nStoreToSave].dpName, kp->Keyboard->dpStoreArray[nStoreToSave].dpString);
   }
 }
+
+void LoadKeyboardOptionsREGCore(LPINTKEYBOARDINFO kp, km_kbp_state* const state)
+{
+  SendDebugMessageFormat(0, sdmKeyboard, 0, "LoadKeyboardOptionsREGCore: Enter");
+  IntLoadKeyboardOptionsCore(REGSZ_KeyboardOptions, kp, state);
+}
+
+BOOL IntLoadKeyboardOptionsCore(LPCSTR key, LPINTKEYBOARDINFO kp, km_kbp_state* state)
+{
+  assert(key != NULL);
+  assert(kp != NULL);
+
+  RegistryReadOnly r(HKEY_CURRENT_USER);
+  if (r.OpenKeyReadOnly(REGSZ_KeymanActiveKeyboards) && r.OpenKeyReadOnly(kp->Name) && r.OpenKeyReadOnly(key))
+  {
+    WCHAR buf[256];
+    int n = 0;
+    // Get the list of default options to determine size of list
+    const km_kbp_keyboard_attrs* keyboardAttrs;
+    km_kbp_status err_status = km_kbp_keyboard_get_attrs(kp->lpCoreKeyboard, &keyboardAttrs);
+    if (err_status != KM_KBP_STATUS_OK) {
+      SendDebugMessageFormat(
+          0, sdmKeyboard, 0, "LoadKeyboardOptionsREGCore: km_kbp_keyboard_get_attrs failed with error status [%d]", err_status);
+    }
```

Comment:
This needs to `return FALSE` if `err_status != KM_KB_STATUS_OK` as `keyboardAttrs` will be in a suspect state.